### PR TITLE
Remove deprecated hasDataAccess method.

### PR DESCRIPTION
**Pull request by abhioncbr**
**Labels**
- cleanup

**Description**
In the year 2021, We deprecated the hasDataAccess method of class [AccessControl.java](https://github.com/apache/pinot/pull/13574/files#diff-1fd3c5096eb98bad72d30be8f0fea8005fa0aa8b78bfceb643fcff2fa009e50e) ( for reference: [PR ](https://github.com/apache/pinot/pull/6507)). This PR is just removing the deprecated method from the code.

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
cc @soumitra-st @zhtaoxiang Can you help take a look?

**Comment by siddharthteotia** [comment]
@vvivekiyer  - Can you help review this please

**Review by vvivekiyer — APPROVED** [review]

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java [hunk 1]
```diff
@@ -158,7 +158,7 @@ public Response downloadSegment(
     boolean hasDataAccess;
     try {
       AccessControl accessControl = _accessControlFactory.create();
-      hasDataAccess = accessControl.hasDataAccess(httpHeaders, tableName);
+      hasDataAccess = accessControl.hasAccess(tableName, AccessType.READ, httpHeaders, null);
```
Comment (vvivekiyer) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java:161]: @abhioncbr Can we please pass the endpointUrl (/methodName) here and in PinotQueryResource (ideally we should pass it in all places where `hasAccess` is called). 

Internally at Linkedin, we have different restrictions on READ access for downloadSegment and query endpoints.

**Review by vvivekiyer — CHANGES_REQUESTED** [review]
Requested changes to pass endpointUrl when calling hasAccess method.

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java (hunk 1 above)
Comment (abhioncbr) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java:161]: Sure, I'll make the required changes. Thanks

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java [hunk 2]
```diff
@@ -292,7 +293,7 @@ private String getQueryResponse(String query, @Nullable SqlNode sqlNode, String
 
     // Validate data access
     AccessControl accessControl = _accessControlFactory.create();
-    if (!accessControl.hasDataAccess(httpHeaders, rawTableName)) {
+    if (!accessControl.hasAccess(rawTableName, AccessType.READ, httpHeaders, Actions.Table.DOWNLOAD_SEGMENT)) {
```
Comment (soumitra-st) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java:296]: Is Actions.Table.DOWNLOAD_SEGMENT is not suited for endpointUrl? If endpointUrl is not accessible, we can use null.

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java [hunk 3]
```diff
@@ -158,7 +158,7 @@ public Response downloadSegment(
     boolean hasDataAccess;
     try {
       AccessControl accessControl = _accessControlFactory.create();
-      hasDataAccess = accessControl.hasDataAccess(httpHeaders, tableName);
+      hasDataAccess = accessControl.hasAccess(tableName, AccessType.READ, httpHeaders, Actions.Table.DOWNLOAD_SEGMENT);
```
Comment (soumitra-st) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java:161]: Same here, Actions.Table.DOWNLOAD_SEGMENT action used as endpointUrl. Shall we pass null?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java (hunk 2 above)
Comment (abhioncbr) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java:296]: We need some consensus on whether to use `Actions.Table.DOWNLOAD_SEGMENT` or pass null.

Comment (vvivekiyer) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java:296]: @soumitra-st Passing endpointUrl is critical to allow implementations to provide different access restrictions to different read endpoints. Hence I had suggested that we add the relevant action. 

All over our codebase (eg: PinotTableResstletResource), we pass the action  as endpointUrl. If we feel that a better identifier is needed to identify endpointUrl, we can discuss. 

@abhioncbr as this is the query endpoint, the action here should be `Actions.Table.QUERY` ?

Comment (soumitra-st) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java:296]: > All over our codebase (eg: PinotTableResstletResource), we pass the action as endpointUrl. If we feel that a better identifier is needed to identify endpointUrl, we can discuss.
@vvivekiyer , if we pass action as endpointUrl across the board, then I am ok with that.

Comment (abhioncbr) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java:296]: Thanks for the feedback. I will make the change here as suggested by @vvivekiyer and for adding endpointUrl across board, I will raise another PR. Sounds good?

**Review by vvivekiyer — APPROVED** [review]

**Comment by abhioncbr** [comment]
Can we merge it?

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessControl.java [hunk 4]
```diff
@@ -55,10 +40,8 @@ default boolean hasDataAccess(HttpHeaders httpHeaders, String tableName) {
    * @param endpointUrl the request url for which this access control is called
    * @return whether the client has permission
    */
-  default boolean hasAccess(@Nullable String tableName, AccessType accessType, HttpHeaders httpHeaders,
-      @Nullable String endpointUrl) {
-    return true;
-  }
+  boolean hasAccess(@Nullable String tableName, AccessType accessType, HttpHeaders httpHeaders,
```
Comment (jadami10) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessControl.java:43]: Why does this no longer default `true`? This is backwards incompatible for users of this interface

Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessControl.java:43]: Good catch. Adding it back in #14382

**Pull request merged**